pkg/ring/ring.go

```diff
@@ -335,20 +338,49 @@ func (r *Ring) GetAll(op Operation) (ReplicationSet, error) {
 	maxUnavailable := r.cfg.ReplicationFactor / 2
 	numRequired -= maxUnavailable
 
-	ingesters := make([]IngesterDesc, 0, len(r.ringDesc.Ingesters))
+	maxUnavailableZones := len(r.ringZones) - maxUnavailable - 1
```

Comment:
I think I've deceived you in my previous comment, showing you a *wrong* pseudo code. I think it should be `maxUnavailableZones := maxUnavailable`. Am I missing anything?

Reply:
No, that's what it comes down to :)